examples/nemo/scripts/utils.py

```diff
@@ -240,9 +238,9 @@ def insert_explicit_loop_parallelism(
         # In addition, they often nest ice linearised loops (npti)
         # which we'd rather parallelise
         if ('ice' in routine_name
-            and isinstance(loop.stop_expr, BinaryOperation)
-            and (loop.stop_expr.operator == BinaryOperation.Operator.UBOUND or
-                 loop.stop_expr.operator == BinaryOperation.Operator.SIZE)
+            and isinstance(loop.stop_expr, IntrinsicCall)
+            and (loop.stop_expr.intrinsic == IntrinsicCall.Intrinsic.UBOUND or
```

Comment:
I shorten it (but I have not used inquiry filter because this was to select particular loops in the ICE routines and we would need to look closely with the profiler the effects of any semantic change we do here - which is probably not for this PR)